### PR TITLE
Oauth Wrapping

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

File: modules/apps/jaggery/mdm/src/mdm/modules/device.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ deviceModule = function () {
             log.error("User object was not found in the session");
             throw constants.ERRORS.USER_NOT_FOUND;
         }
```
Comment (harshanL) [inline modules/apps/jaggery/mdm/src/mdm/modules/device.js]: Why there are error logs in finally blocks? and it seems like there is something wrong with the logic here.

Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/modules/device.js]: There is no error logs in finally blocks in this PR.
I think you have mistaken becouse some of the cde is shrink.Click the expand icon on the left side to get the full code.

File: modules/apps/jaggery/mdm/src/mdm/units/service-invoker-utility/public/js/invoker-lib.js [hunk 2]
```diff
@@ -52,7 +52,12 @@ var invokerUtil = function () {
             }
             $.ajax(data).fail(function (jqXHR) {
                 if (jqXHR.status == "401"){
-                    window.location.replace("/mdm");
+                    requestAccessToken(function () {
+                        execute();
+                    },function () {
```
Comment (prabathabey) [inline modules/apps/jaggery/mdm/src/mdm/units/service-invoker-utility/public/js/invoker-lib.js]: Formatting issues.

Comment (Kamidu) [inline modules/apps/jaggery/mdm/src/mdm/units/service-invoker-utility/public/js/invoker-lib.js]: These formatting issues has been fixed in a recent PR

**Pull request merged**